src/puppetlabs/jdbc_util/core.clj

```diff
@@ -318,15 +318,19 @@
   If the column has no associated sequence, throws an Exception."
   [db table column]
   (if-let [sequence-name (get-sequence-name-for-column db table column)]
-    (let [select-last-value (str "(SELECT last_value FROM " (quoted sequence-name) ")")
+    (let [select-is-called (str "(SELECT is_called FROM " (quoted sequence-name) ")")
+          select-seq-values (str "(SELECT last_value, is_called FROM " (quoted sequence-name) ")")
           select-max-in-column (str "(SELECT MAX(" (quoted column) ") FROM " (quoted table) ")")]
       (jdbc/with-db-transaction [txn-db db]
         (jdbc/execute! txn-db [(format "LOCK TABLE \"%s\" IN EXCLUSIVE MODE" table)])
         (jdbc/query txn-db [(str "SELECT"
                                  "  CASE"
-                                 "  WHEN " select-max-in-column " > " select-last-value
-                                 "    THEN setval('" sequence-name "', " select-max-in-column ")"
-                                 "  END")])))
+                                 "  WHEN seq.is_called = true AND " select-max-in-column " > seq.last_value"
+                                 "    THEN setval('" sequence-name "', " select-max-in-column ", true)"
+                                 "  WHEN seq.is_called = false AND " select-max-in-column " >= seq.last_value"
```

Comment:
>  Now when is_called is false, and the max row value and last_value are equal, is_called will be set to true in order to get the next value that will can be stored in the table.

Based on that part of the commit comment, it sounds like the value of `is_called` is supposed to change. Does it change when the sequence value is updated, or is there an update to `is_called` missing?

Reply:
Cool cool, I thought it might be the case that something updated `is_called`. Just wanted to double check!